lib/elixir/lib/module/types/helpers.ex

```diff
@@ -70,6 +70,21 @@ defmodule Module.Types.Helpers do
         "var.fun()" (with parentheses) means "var" is an atom()
         """
 
+      :interpolation ->
+        """
+
+        #{hint()} string interpolation in Elixir uses the String.Chars protocol to \
+        convert a data structure into a string. Either convert the data type into a \
+        string upfront or implement the protocol accordingly
```

Comment:
Should we suggest *for which type* to implement this?

Reply:
Unfortunately no, because there may be an infinite amount of them. If someone has a type of `not Range.t`, it can be any other struct, which wouldn't be valid in String.Chars (and some structs definitely do not implemnt `String.Chars` , but we also cannot list them).